### ISSUE TITLE
synthesizeToFile cannot run concurrently

### ISSUE THREAD

**Issue by WillowWisp** [dlutton/flutter_tts#272]
## 🐛 Bug Report
When running `synthesizeToFile` concurrently to generate multiple files at once, only the first file is generated. This happens on both Android and iOS

### Expected behavior
`synthesizeToFile` should be allowed to run concurrently to reduce the time generating multiple files at once.

### Reproduction steps
```
final FlutterTts _flutterTts = FlutterTts();

await _flutterTts.awaitSynthCompletion(true);

await _flutterTts.setLanguage('en-US');

await Future.wait([
  _flutterTts.synthesizeToFile(
    "Lorem ipsum is a name for a common type of placeholder text. Also known as filler or dummy text, this is simply copy that serves to fill a space without actually saying anything meaningful. It's essentially nonsense text, but gives an idea of what real words will look like in the final product.",
    'file1.wav',
  ),
  _flutterTts.synthesizeToFile(
    "The main reason why lorem ipsum text is used is that it keeps people from focusing on the actual text. When someone creates a template and asks for feedback on it, they don't want the people reviewing it to get distracted by what the text says.",
    'file2.wav',
  ),
]);

// After this, only file1.wav would be generated
```

Maybe this has something to do with the `awaitSynthCompletion` function but I'm not sure.

**Platform:** 
- [x] :iphone: iOS
- [x] :robot: Android

**Comment by dlutton** [comment]
@WillowWisp unfortunately it's not meant to be ran concurrently, you'd have to await each one individually.

**Comment by WillowWisp** [comment]
That's really unfortunate 😟. Is it because of the native TTS?

**Comment by dlutton** [comment]
@WillowWisp most TTS services aren't meant to run concurrently, at least I can't think of a situation where you'd want to play TTS concurrently.  SynthesizeToFile is probably the only thing that makes sense to run concurrently for TTS.  I'll have to research to see if I can add this enhancement and add support for both iOS and Android.

**Comment by WillowWisp** [comment]
@dlutton
> SynthesizeToFile is probably the only thing that makes sense to run concurrently for TTS

Yeah, that's exactly my case. Hopefully, there would be a way to fix this because having to run `synthesizeToFile` sequentially + #271 makes the experience on iOS a lot worse 😟.
Anyway, thank you for your great support & effort on the package, Rick 😄
